### PR TITLE
Add release post endpoint

### LINKED ISSUE #948: Backend: Create API endpoints for promoting/releasing revisions

### PR DESCRIPTION
**Pull request by tbille**
# Summary
Fixes #948 
Add post release endpoint
I had to adapt the BaseTests to be able to have some api style endpoints

# QA

@bartaz you can try to rebase your [PR](https://github.com/canonical-websites/snapcraft.io/pull/1059) on this to try the data you are posting
It uses and returns the same info that the api: https://dashboard.snapcraft.io/docs/api/snap.html#release-a-snap-build-to-a-channel

### CODE REVIEW

**Comment by bartaz** [comment]
@tbille It would be good if we could make sure errors are returned as JSON. Currently errors seems to be returned same as on other pages, as HTML, so for example a missing field error returns:
```
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2 Final//EN">
<title>502 Bad Gateway</title>
<h1>Bad Gateway</h1>
<p>missing-field</p>
```
and store API docs say it would be JSON response: https://dashboard.snapcraft.io/docs/api/snap.html#release-a-snap-build-to-a-channel

**Comment by bartaz** [comment]
When there are no errors API works correctly, release is done in the store and updated channel map is returned.

File: tests/publisher/snaps/tests_post_release.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        endpoint_url = "/account/snaps/{}/release".format(snap_name)
+
+        super().setUp(
+            snap_name=snap_name,
+            endpoint_url=endpoint_url,
+            method_endpoint="POST",
+        )
+
+
+class PostDataReleasePage(BaseTestCases.EndpointLoggedIn):
```
Comment (bartaz) [inline tests/publisher/snaps/tests_post_release.py]: Would be good to have a test than errors return as JSON?

Comment (tbille) [reply tests/publisher/snaps/tests_post_release.py]: done

**Pull request merged**